### PR TITLE
Version Patch v0.0.2

### PR DESCRIPTION
**Pull request by GrayHatter**
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/202)

### CODE REVIEW

**Comment by GrayHatter** [comment]
suggested by @nurupo because the logger was fixed in 949ef785a4e7aa7868c9605b6bbed15c3f4beab9 that would be a good place to increment the patch version.

**Comment by iphydf** [comment]
This incorrectly passes tests. The tests are broken, I'll fix them.

**Comment by GrayHatter** [comment]
blocked by #204 ping me on merge so I can rebase.

**Comment by iphydf** [comment]
@GrayHatter ping: #204 is merged, please rebase.

**Comment by GrayHatter** [comment]
@iphydf done

**Comment by nurupo** [comment]
@GrayHatter at the very least you have missed https://github.com/TokTok/c-toxcore/blob/master/CMakeLists.txt#L9. Please check the coomit that did 0.0.1 version tick and see if anything esle is missing.

**Comment by dvor** [comment]
I don't like that version number is spread all over the project. Is it possible to keep it in single place?

People will change version every release, and people _will_ forget to change it somewhere in those files.

Also even having it in multiple places, you cannot simply search for "0.0.1" string as it is broken into separate major, minor, patch components.

**Comment by nurupo** [comment]
@dvor there is already a script for updating versions in all the places, I'm not sure why @GrayHatter is not using it. All you need to do is bump the version in CMakeLists.txt and it will run https://github.com/TokTok/c-toxcore/blob/master/other/version-sync https://github.com/TokTok/c-toxcore/blob/master/CMakeLists.txt#L458 which should update the version number everywhere.

File: other/version-sync [hunk 1]
```diff
@@ -26,3 +26,7 @@ update 'configure.ac' 's/AC_INIT(\[tox\], \[.*\])/AC_INIT([tox], ['$VER'])/'
 update 'toxcore/tox.api.h' 's/\(const VERSION_MAJOR *= \).*;/\1'$MAJOR';/'
 update 'toxcore/tox.api.h' 's/\(const VERSION_MINOR *= \).*;/\1'$MINOR';/'
 update 'toxcore/tox.api.h' 's/\(const VERSION_PATCH *= \).*;/\1'$PATCH';/'
+
+update 'CMakeLists.txt' 's/^\(set(PROJECT_VERSION_MAJOR \"\).*")/\1'$MAJOR'")/'
+update 'CMakeLists.txt' 's/^\(set(PROJECT_VERSION_MINOR \"\).*")/\1'$MINOR'")/'
+update 'CMakeLists.txt' 's/^\(set(PROJECT_VERSION_PATCH \"\).*")/\1'$PATCH'")/'
```
Comment (nurupo) [inline other/version-sync]: Uh, you are not supposed to call this script manually. It's supposed to be called by cmake. When releasing a new version, you update the version number in the CMakeLists.txt and run cmake, which will update the version everywhere. At least that'she how I understand it's supposed to be done.

Comment (iphydf) [inline other/version-sync]: Yes, that's correct. And failing to do so should fail travis.

**Comment by GrayHatter** [comment]
Wait... So cmake is now a dependency for autotools?

Lolwut

Okay I'll fix it tonight. Anyone else can feel free to beat me

**Comment by iphydf** [comment]
No, cmake is not a dependency. The source of truth for the version number is in the cmake file, but that is only relevant when changing the number.

**Comment by iphydf** [comment]
:lgtm_strong:

---

Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/202#-:-KVLoINphdVDHbOM8BDf:b-jbsf8z)*

**Pull request merged**